examples/src/main/resources/precompile-example/PrecompileExample.sol

```diff
@@ -0,0 +1,237 @@
+// and copy the outputted json file to ./PrecompileExample.json
+
+contract PrecompileExample is ExpiryHelper, PrngSystemContract {
+    address payable owner;
+    address payable aliceAccount;
+    address fungibleToken;
+    address nftToken;
+
+    constructor(address payable _owner, address payable _aliceAccount) {
+        owner = _owner;
+        aliceAccount = _aliceAccount;
+    }
+
+    function step0() external returns (bytes32 result) {
+        require(msg.sender == owner);
+
+        result = this.getPseudorandomSeed();
+    }
+
+    // In order for some functions (such as createFungibleToken) to work, the contract must possess the funds for
+    // the function call.  We are using ContractExecuteTransaction.setPayableAmount() to transfer some Hbar
+    // to the contract's account at each step (which means this function must be payable), and then transferring
+    // the excess Hbar back to the owner at the end of each step.
+    function step1() external payable returns (int responseCode) {
+        require(msg.sender == owner);
+
+        IHederaTokenService.TokenKey[] memory keys = new IHederaTokenService.TokenKey[](1);
+        // Set the admin key, supply key, pause key, and freeze key to the key of the account that executed function (INHERIT_ACCOUNT_KEY).
+        keys[0] = createSingleKey(ADMIN_KEY_TYPE | SUPPLY_KEY_TYPE | PAUSE_KEY_TYPE | FREEZE_KEY_TYPE, INHERIT_ACCOUNT_KEY, bytes(""));
+
+        (responseCode, fungibleToken) = createFungibleToken(
+            IHederaTokenService.HederaToken(
+                "Example Fungible token", // name
+                "E", // symbol
+                owner, // treasury
+                "memo",
+                true, // supply type, false -> INFINITE, true -> FINITE
+                1000, // max supply
+                false, // freeze default (setting to false means that this token will not be initially frozen on creation)
+                keys, // the keys for the new token
+            // auto-renew fee paid by aliceAccount every 7,000,000 seconds (approx. 81 days).
+            // This is the minimum auto renew period.
+                createAutoRenewExpiry(aliceAccount, 7000000)
+            ),
+            100, // initial supply
+            0 // decimals
+        );
+
+        // send any excess Hbar back to the owner
+        owner.transfer(address(this).balance);
+    }
+
+    function step2() external returns (int responseCode) {
+        require(msg.sender == owner);
+
+        uint64 newTotalSupply;
+        int64[] memory mintedSerials; // applicable to NFT tokens only
+        (responseCode, newTotalSupply, mintedSerials) = mintToken(
+            fungibleToken,
+            100, // amount (applicable to fungible tokens only)
+            new bytes[](0) // metadatas (applicable to NFT tokens only)
+        );
+
+        require(newTotalSupply == 100 + 100);
+    }
+
+    function step3() external returns (int responseCode) {
+        require(msg.sender == owner);
+
```

Comment:
You can associate the token to the contract address directly here in the contract as a quick fix suggestion to the issue with step5, e.g.  responseCode = associateToken(address(this), fungibleToken) .